src/controller/MazeController.java

```diff
@@ -0,0 +1,68 @@
+package controller;
+
+import model.MazeModel;
+import view.CLI;
+
+import java.io.IOException;
+
+public class MazeController {
+
+    private CLI cli_view;
+    private MazeModel maze_model;
+
+    public Observable VCObservable;
+    public Observer CVObserver;
+
+    public Observable CMObservable;
+    public Observer MCObserver;
+
+    public Observable MCObservable;
+    public Observer CMObserver;
+
+    public Observable CVObservable;
+    public Observer VCObserver;
+
+    public MazeController(CLI cli_view, MazeModel maze_model) {
+        this.cli_view = cli_view;
+        this.maze_model = maze_model;
+
+        this.VCObservable = new Observable();
+        this.CVObserver = new Observer();
+        VCObservable.add(CVObserver);
+
+        this.CMObservable = new Observable();
+        this.MCObserver = new Observer();
+        CMObservable.add(MCObserver);
+
+        this.MCObservable = new Observable();
+        this.CMObserver = new Observer();
+        MCObservable.add(CMObserver);
+
+        this.CVObservable = new Observable();
+        this.VCObserver = new Observer();
+        CVObservable.add(VCObserver);
+    }
+
+    @SuppressWarnings("InfiniteLoopStatement")
```

Comment:
Even if it's auto-complete, it is great!

Reply:
I added it due to a warning while committing :)

Reply:
Hope you understand why I see it so good and helping